src/libsystemd-network/sd-dhcp6-client.c

```diff
@@ -198,6 +198,26 @@ int sd_dhcp6_client_set_duid(sd_dhcp6_client *client, uint16_t duid_type,
         return 0;
 }
 
+int sd_dhcp6_client_get_duid(sd_dhcp6_client *client, uint16_t *duid_type,
+                             const uint8_t **duid, size_t *duid_len) {
+        assert_return(client, -EINVAL);
+        assert_return(duid_type, -EINVAL);
+        assert_return(duid_len, -EINVAL);
+        assert_return(duid, -EINVAL);
```

Comment:
Shouldn't NULL be accepted here? E.g. if duid_type == NULL, do nothing, otherwise fill it.